etlman/projects/tests/test_views/test_view_pipeline.py

```diff
@@ -422,3 +422,17 @@ def test_back_edit_step_post(self, nonadmin_client, project):
         assert step_session_key in nonadmin_client.session
         assert nonadmin_client.session[step_session_key]["name"] == data["name"]
         assert pipeline.name in str(response.content)
+
+
+@pytest.mark.django_db
+class TestClearSession:
+    def test_cancel_new_step_get(self, nonadmin_client, project):
+        response = nonadmin_client.get(
```

Comment:
This test is not really testing the functionality of the view. Proposed improvements:

- [ ] Call `save_step2_data_in_session()` to save some values to the session. Maybe started with unsaved (`.build()`) model objects (including `Pipeline`, `DataInterface`, and `Step`)
- [ ] Call `nonadmin_client.get()` to execute the view
- [ ] Inspect `nonadmin_client.session` to ensure it no longer contains the keys in question
- [ ] Create a second test that follows the same pattern and works with model objects that have been saved to the DB (i.e., created with a factory _without_ the `.build()` suffix)

Reply:
I think all suggestions are addressed.